app/authenticators/osf-cookie.ts

```diff
@@ -18,60 +20,62 @@ import authenticatedAJAX from 'ember-osf-web/utils/ajax-helpers';
             // Note: we have to deepcopy res because pushPayload mutates our data
             // and causes an infinite loop because reasons
-            this.get('store').pushPayload(Ember.copy(res, true));
+            this.get('store').pushPayload(copy(res, true));
             return res.data;
         });
-    },
+    }
+
     restore(/* data */) {
-        return this._test().fail(this.invalidate);
-    },
+        return this._test().catch(this.invalidate);
+    }
 
     /**
      * Send a request to the flask application to trigger invalidation of session remotely
      * @method invalidate
      */
-    invalidate(data) {
+    invalidate(this: OsfCookieAuthenticator, data) {
         if ((!data || data.id) && data.status !== 401) {
-            // If invalidate is called when loading the page to check if a cookie has permissions, don't redirect the user
-            // But if invalidate is called without arguments, or for any other reason, interpret this as a straight logout request
-            //   (and let the session get invalidated next time at the start of first page load)
+            // If invalidate is called when loading the page to check if a
+            // cookie has permissions, don't redirect the user.
+            //
+            // But if invalidate is called without arguments, or for any other reason,
+            // interpret this as a straight logout request. Let the session get invalidated
+            // next time at the start of first page load.
+            //
             // Can't do this via AJAX request because it redirects to CAS, and AJAX + redirect = CORS issue
 
-            // Manually clear session before user leaves the page, since we aren't sticking around for ESA to do so later
-            return this.get('session.session')._clear(true);
+            // Manually clear session before user leaves the page, since
+            // we aren't sticking around for ESA to do so later
+            return this.get('session').get('session')._clear(true);
         } else {
-            // This branch is expected to be called when a test request reveals the user to lack permissions... so session should be wiped
-            return Ember.RSVP.resolve();
+            // This branch is expected to be called when a test request reveals
+            // the user to lack permissions... so session should be wiped
+            return resolvedPromise();
         }
-    },
+    }
+
     /**
      * For now, simply verify that a token is present and can be used
      * @method authenticate
      * @param code
      * @return {Promise}
      */
-    authenticate(code) {
-        // NOTE: Must be wrapped in an RSVP promise
-        // _test returns a Jquery Promise but authenticate expects an RSVP Promise
-        return new Ember.RSVP.Promise((resolve, reject) =>
-            this._test(code).then(resolve).fail(reject));
-    },
-});
+    authenticate() {
+        return this._test();
```

Comment:
So id doesn't actually have to be wrapped in an RSVP promise?

Reply:
I [changed `authenticatedAJAX` to take care of that](https://github.com/CenterForOpenScience/ember-osf-web/pull/137/files#diff-dc071bc0df9a4a5963ad9ad7af84f02bR36), since it was the source of the jQuery promise. Now no one else has to worry.